src/sha512.rs

```diff
@@ -78,6 +78,32 @@ impl PartialEq for Hash {
 
 impl Eq for Hash {}
 
+impl Default for Hash {
+    fn default() -> Hash {
+        Hash([0; 64])
+    }
+}
+
+use std::cmp::Ordering;
+
+impl PartialOrd for Hash {
+    fn partial_cmp(&self, other: &Hash) -> Option<Ordering> {
+        (&self).partial_cmp(&other)
```

Comment:
wat. I'm confused.

Reply:
Oh, maybe make the slice conversion clear?

Reply:
@TheBlueMatt Eh... Like `(&self[..])`? Kinda defeats the whole purpose of https://github.com/rust-bitcoin/bitcoin_hashes/pull/13

Reply:
I mean in one case its clear, in this case someone (like me) who doesn't know rust as well could *totally* see the compiler just calling partial_cmp recursively directly here...I'd also accept (&self.0).partial_cmp(&other.0).